lib/tasks_private/terraform/common.rb

```diff
@@ -0,0 +1,77 @@
+require "json"
+require 'yaml'
+require 'logger'
+require 'ruby-terraform'
+
+def setup
+  @current_dir = ENV['cwd'] || Dir.pwd
+  @setting_dir = File.join(@current_dir, "lib/tasks_private/terraform")
+  @test_dir = File.join(@current_dir, "lib/tasks_private/terraform/#{@testname}")
+  Dir.chdir(@test_dir)
+
+  log_file = Logger::LogDevice.new('ruby_terraform.log')
+  stdout_device = Logger::LogDevice.new($stdout)
+  multi_io = RubyTerraform::MultiIO.new(log_file, stdout_device)
+  logger = Logger.new(multi_io)
+
+  RubyTerraform.configure do |tconfig|
+    tconfig.logger = logger
+    tconfig.stdout = multi_io
+    tconfig.stderr = multi_io
+  end
+  RubyTerraform.init
+
+  secrets = YAML.safe_load(File.read(File.join(@current_dir, "config/secrets.yml")), :aliases => true)
+  settings = YAML.safe_load(File.read(File.join(@setting_dir, "powervs_testsettings.yml")), :aliases => true)
+
+  setting = settings["test"].find { |key| key["testname"] == @testname }
+  {:secret => secrets["test"]["ibm_cloud_power"], :setting => setting}
+end
+
+def plan_vars(config)
+  settings = {
+    :ibmcloud_api_key    => config[:secret]["api_key"],
+    :power_instance_id   => config[:secret]["cloud_instance_id"],
+    :ibmcloud_region     => config[:secret]["ibmcloud_region"],
+    :vm_name             => config[:setting]["vm_name"],
+    :volume_name         => config[:setting]["volume_name"],
+    :key_pair_name       => config[:setting]["key_pair_name"],
+    :image_name          => config[:setting]["image_name"],
+    :sys_type            => config[:setting]["sys_type"],
+    :public_network_name => config[:setting]["public_network_name"],
+    :power_network_name  => config[:setting]["power_network_name"]
+  }
+  settings
```

Comment:
Hey you actually don't need to set a local variable if you just want to return the thing, ruby will return the last return anyway and declaring the hash will return itself:
```suggestion
  {
    :ibmcloud_api_key    => config[:secret]["api_key"],
    :power_instance_id   => config[:secret]["cloud_instance_id"],
    :ibmcloud_region     => config[:secret]["ibmcloud_region"],
    :vm_name             => config[:setting]["vm_name"],
    :volume_name         => config[:setting]["volume_name"],
    :key_pair_name       => config[:setting]["key_pair_name"],
    :image_name          => config[:setting]["image_name"],
    :sys_type            => config[:setting]["sys_type"],
    :public_network_name => config[:setting]["public_network_name"],
    :power_network_name  => config[:setting]["power_network_name"]
  }
```

Reply:
Ha I beat rubocop to it :wink: https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/382#issuecomment-1130235110